ContactPoint/Services/AutoAnswerService.cs

```diff
@@ -118,13 +116,10 @@ void _timer_Elapsed(object sender, ElapsedEventArgs e)
 
         bool CanPickupCall(ICall callToPickup)
         {
-            foreach (var call in Core.CallManager)
-                if (call != callToPickup)
-                    if (call.State == CallState.ACTIVE || call.State == CallState.HOLDING || // Check if we have an active call
-                        call.State == CallState.CONNECTING || call.State == CallState.ALERTING) // Check if we have an outgoing call
-                        return false;
-
-            return true;
+            return !Core.CallManager.ToArray().Any(call => 
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Drop this useless call to 'ToArray' or replace it by 'AsEnumerable' if you are using LINQ to Entities. ([csharpsquid:S2971](https://sonarcloud.io/organizations/alex-yevtushenko3820/rules#rule_key=csharpsquid%3AS2971))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=IPP&pullRequest=7&issues=AWfG1-fh8DroMTOoScWk&open=AWfG1-fh8DroMTOoScWk)

[ISSUE_KEY]: AWfG1-fh8DroMTOoScWk